### DISCUSSION TITLE
Static Site Generation based on Vercel KV data

### DISCUSSION THREAD

**Discussion by jschuur** [vercel/next.js#49328] [App Router] [answered]
How could I statically generate a route based on Vercel KV data if I'm using the app directory (where I can't do `getStaticProps`)?

For fetch request data, that seems to be [static at build time](https://vercel.com/blog/nextjs-app-router-data-fetching#static-vs.-dynamic-data), but a KV `.get` call doesn't seem to have any [cache options](https://vercel.com/docs/storage/vercel-kv/kv-reference#get).

I could make a fetch request to a route that executes the kv `.get` command, which would be cached. Assuming a site can fetch itself during build time (or I just hit the previously deployed version of the site).

Perhaps mine is a weird use case: I'm generating KV data from a cron job that executes a long running query every hour and it sounded like bundling the KV cached data with the build would improve performance. Plus it keeps down the number of KV requests, considering the [limits](https://vercel.com/docs/storage/vercel-kv/usage-and-pricing#pricing).

**Comment by dravorle** [comment]
You could try using the `cache` function provided by `react`. This seems to be the way to cache stuff that does not use the fetch-function.

Example from the [Docs](https://nextjs.org/docs/app/building-your-application/data-fetching/caching#per-request-caching):
```ts
import { cache } from 'react';
 
export const getUser = cache(async (id: string) => {
  const user = await db.user.findUnique({ id });
  return user;
});
```

If you want to use something like the cache tags that are available within the cache options of the fetch you might have to use the `unstable_cache` function provided by `next/cache` for now.

**Comment by jschuur** [comment, marked as answer]
I've since found out how to [configure route segments](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#revalidate), and that by default it's already going to be static at build time. You'd actively have to use `export const revalidate = 0` to make the route dynamic.
